Windows gradle check failure - checkExtraction tasks failing

On running gradle check on windows, we see the below group of failures for checkExtraction task. We need to modify these extractArchive tasks so they get skipped on windows as these distributions aren't built on windows platform.


> Task :distribution:archives:linux-arm64-tar:checkExtraction FAILED
> Task :distribution:archives:darwin-arm64-tar:checkExtraction FAILED
> Task :distribution:archives:freebsd-tar:checkExtraction FAILED
> Task :distribution:archives:linux-s390x-tar:checkExtraction FAILED
> Task :distribution:archives:darwin-tar:checkExtraction FAILED
> Task :distribution:archives:no-jdk-darwin-arm64-tar:checkExtraction FAILED
> Task :distribution:archives:no-jdk-darwin-tar:checkExtraction FAILED
> Task :distribution:archives:no-jdk-linux-tar:checkExtraction FAILED
> Task :distribution:archives:no-jdk-freebsd-tar:checkExtraction FAILED
> Task :distribution:archives:linux-tar:checkExtraction FAILED

 ./gradlew :distribution:archives:darwin-tar:checkExtraction

> Task :distribution:archives:darwin-tar:checkExtraction FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':distribution:archives:darwin-tar:checkExtraction'.
> Could not read 'C:\Users\Pooj\OpenSearch\distribution\archives\darwin-tar\build\distributions\opensearch-min-3.0.0-SNAPSHOT-darwin-x64.tar.gz' as it does not exist.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -174,7 +174,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - [BUG]: Allow decommission to support delay timeout ([#4930](https://github.com/opensearch-project/OpenSearch/pull/4930))
 - Fix failing test: VerifyVersionConstantsIT ([#4946](https://github.com/opensearch-project/OpenSearch/pull/4946))
 - Fixed compression support for h2c protocol ([#4944](https://github.com/opensearch-project/OpenSearch/pull/4944))
-
+- Fix for failing checkExtraction, checkLicense and checkNotice tasks for windows gradle check ([#4941](https://github.com/opensearch-project/OpenSearch/pull/4941))
 ### Security
 - CVE-2022-25857 org.yaml:snakeyaml DOS vulnerability ([#4341](https://github.com/opensearch-project/OpenSearch/pull/4341))
 

--- a/buildSrc/src/main/java/org/opensearch/gradle/internal/InternalDistributionArchiveCheckPlugin.java
+++ b/buildSrc/src/main/java/org/opensearch/gradle/internal/InternalDistributionArchiveCheckPlugin.java
@@ -73,12 +73,14 @@ public class InternalDistributionArchiveCheckPlugin implements Plugin<Project> {
             .create("distributionArchiveCheck", DistributionArchiveCheckExtension.class);
 
         File archiveExtractionDir = calculateArchiveExtractionDir(project);
-
         // sanity checks if archives can be extracted
         TaskProvider<Copy> checkExtraction = registerCheckExtractionTask(project, buildDistTask, archiveExtractionDir);
+        checkExtraction.configure(InternalDistributionArchiveSetupPlugin.configure(buildTaskName));
         TaskProvider<Task> checkLicense = registerCheckLicenseTask(project, checkExtraction);
+        checkLicense.configure(InternalDistributionArchiveSetupPlugin.configure(buildTaskName));
 
         TaskProvider<Task> checkNotice = registerCheckNoticeTask(project, checkExtraction);
+        checkNotice.configure(InternalDistributionArchiveSetupPlugin.configure(buildTaskName));
         TaskProvider<Task> checkTask = project.getTasks().named("check");
         checkTask.configure(task -> {
             task.dependsOn(checkExtraction);
@@ -118,7 +120,7 @@ public class InternalDistributionArchiveCheckPlugin implements Plugin<Project> {
     }
 
     private TaskProvider<Task> registerCheckLicenseTask(Project project, TaskProvider<Copy> checkExtraction) {
-        TaskProvider<Task> checkLicense = project.getTasks().register("checkLicense", task -> {
+        return project.getTasks().register("checkLicense", task -> {
             task.dependsOn(checkExtraction);
             task.doLast(new Action<Task>() {
                 @Override
@@ -138,7 +140,6 @@ public class InternalDistributionArchiveCheckPlugin implements Plugin<Project> {
                 }
             });
         });
-        return checkLicense;
     }
 
     private TaskProvider<Copy> registerCheckExtractionTask(Project project, TaskProvider<Task> buildDistTask, File archiveExtractionDir) {

--- a/buildSrc/src/main/java/org/opensearch/gradle/internal/InternalDistributionArchiveSetupPlugin.java
+++ b/buildSrc/src/main/java/org/opensearch/gradle/internal/InternalDistributionArchiveSetupPlugin.java
@@ -87,7 +87,7 @@ public class InternalDistributionArchiveSetupPlugin implements Plugin<Project> {
         configureTarDefaults(project);
     }
 
-    private Action<Task> configure(String name) {
+    static Action<Task> configure(String name) {
         return (Task task) -> task.onlyIf(s -> {
             if (OperatingSystem.current().isWindows()) {
                 // On Windows, include only Windows distributions and integTestZip

